controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -2079,6 +2079,11 @@ func (r *BareMetalHostReconciler) updateEventHandler(e event.UpdateEvent) bool {
 		return true
 	}
 
+	// The assumption that this will not always be filled out is why this check is not appended to the below block
+	if oldBMH.Status.LastUpdated != nil && newBMH.Status.LastUpdated != nil && oldBMH.Status.LastUpdated != newBMH.Status.LastUpdated && reflect.DeepEqual(oldBMH.Spec, newBMH.Spec) && reflect.DeepEqual(oldBMH.ObjectMeta, newBMH.ObjectMeta) {
+		return false
+	}
+
 	// Discard updates that did not increase the resource Generation (such as on Status.LastUpdated), except for the finalizers or annotations
 	if reflect.DeepEqual(e.ObjectNew.GetFinalizers(), e.ObjectOld.GetFinalizers()) && reflect.DeepEqual(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations()) {
```

Comment:
Isn't this already a rather awkward way of saying `if reflect.DeepEqual(oldBMH.ObjectMeta, newBMH.ObjectMeta) { return false }`?
i.e. this already captures everything that the new test above would capture, and then some.

Reply:
The above check from this patch would make sure that everything in ObjectMeta is equal including what is not explicitly set with the Finalizers and/or the Annotations, e.g: Labels, OwnerRefs, ResourceVersion, Generation, etc.  

The check on line 2088 seems to only care about the Finalizers and Annotations which would create requeues with Label changes, ownerRefs, etc which would be valid. 

The check I am putting in makes sure that the only change that is different is that the LastUpdated on the status is changed without anything else being different, therefor not requeuing.

Reply:
We basically have:

```
if foo && bar && baz &&quux {
  return false
}
if foo {
  return false
}
return true
```

When put like that, it's pretty easy to see that the first check isn't doing anything.

Reply:
This change is now outdated and only includes the test to verify that the change implemented to address the root issue is resolved. 

The thought behind it was the first check would include the other changes that don’t get captured in the second check which is why the status is not included but only the spec/metadata and the LastUpdated.

Reply:
Resolving as not relevant to the review.